### PR TITLE
Move Workspace creation in the onboardingFlow to OnboardingPurpose page and skip Workspace creation for OD signups

### PR DESCRIPTION
**Pull request by Shahidullah-Muffakir**
### Explanation of Change

With the new BE changes, when a user signs up through OldDot and is redirected to NewDot, the backend now automatically creates a workspace for **vsb and smb** signup qualifiers.

In this PR, we skip the frontend workspace creation for these cases. We also move the workspace creation logic in the onboarding flow from the `BaseOnboardingEmployees` page to the `BaseOnboardingPurpose` page. Since `BaseOnboardingPurpose` is skipped for OD signups, this ensures no duplicate workspace is created.

Additionally, we added two conditions:
hasSignupQualifier && filteredPolicies.length === 0
These prevent edge cases where a workspace might be created twice.

### Fixed Issues

$ https://github.com/Expensify/App/issues/53326
PROPOSAL: https://github.com/Expensify/App/issues/53326#issuecomment-2508586081

### Tests

**Test 1**
1. Sign up through OD.
2. Select **Manage expenses for a small team (1-9 employees)** during signup.
3. After being redirected to ND
4. On the **account page** (Do you use any accounting software? page), select any option and click **Continue**.
5. Verify that the **CreateWorkspace** API is not called during the onboarding flow (since the workspace is now automatically created in the backend).
6. After completing the onboarding flow, confirm that one workspace is successfully created with the new backend changes merged.

**Test 2**

1. Sign up through ND.
2. On the onboarding purpose step, select **Manage my team's expenses.**
3. On the **Employees page** (How many employees do you have? page), select any option and click **Continue**.
4. On the **Accounting page** (Do you use any accounting software? page), select any option and click **Continue**.
5. Verify that the **CreateWorkspace** API is called during the onboarding flow
7. Verify that one workspace is created.

- [x] Verify that no errors appear in the JS console

### Offline tests

Internet connection required.
### QA Steps

**Test 1**
1. Sign up through OD.
2. Select **Manage expenses for a small team (1-9 employees)** during signup.
3. After being redirected to ND
4. On the **Accounting page ** (Do you use any accounting software? page), select any option and click **Continue**.
5. Verify that the **CreateWorkspace** API is not called during the onboarding flow (since the workspace is now automatically created in the backend).
6. After completing the onboarding flow, confirm that one workspace is successfully created with the new backend changes merged.

**Test 2**

1. Sign up through ND.
2. On the onboarding purpose step, select **Manage my team's expenses.**
3. On the **Employees page** (How many employees do you have? page), select any option and click **Continue**.
4. On the **Accounting page** (Do you use any accounting software? page), select any option and click **Continue**.
5. Verify that the **CreateWorkspace** API is called during the onboarding flow
7. Verify that one workspace is created.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/a31794e1-ce9d-4a72-a1bb-c4e14b0ec20c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/3627bf2e-5ed3-4529-8d8a-8b585d3f9f83

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/c0017ef0-0636-4137-98aa-3e002068a331

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/8df47cf3-f7b4-40da-9bbc-1e9a43aaed2c

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Signup through ND:

https://github.com/user-attachments/assets/76971196-4fce-40ea-9d28-0b00407207a8

Signup through OD:

https://github.com/user-attachments/assets/50da45c8-c38e-4266-9c89-5b52c23fd9f4

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/44f9eaec-cb6d-4423-81bb-ffc31e92f377

</details>

### CODE REVIEW

**Comment by carlosmiceli** [comment]
@thesahindia this one is kinda urgent, in case you can prioritize it. Thank you! 🙏

**Review by thesahindia — APPROVED** [review]
LGTM!

File: src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx
Comment (marcaaron) [inline src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx:98]: I'm confused about why we moved this here.

**Review by marcaaron — CHANGES_REQUESTED** [review]

**Comment by marcaaron** [comment]
>the backend now automatically creates a workspace for vsb and smb signup qualifiers

Just for clarity, it **does not do this yet**. This PR will need to support the existing flow AND that change when it happens?

>We also move the workspace creation logic in the onboarding flow from the BaseOnboardingEmployees page to the BaseOnboardingPurpose page. Since BaseOnboardingPurpose is skipped for OD signups, this ensures no duplicate workspace is created.

I think we have to unwind this part. I believe it could temporarily reintroduce [this bug](https://github.com/Expensify/App/issues/50758). And, yes, the BE PRs will _eventually_ fix it, but the plan was to deploy them this PR is deployed. 

So, App needs to exist in a state where it will maybe create a policy for now and then stop creating one after the backend creates one. At least, I thought that was the plan.

@carlosmiceli Let me know if I have this right or if I'm missing something...

**Comment by carlosmiceli** [comment]
@marcaaron agreed on everything but this line:

> App needs to exist in a state where it will maybe create a policy for now and then stop creating one after the backend creates one.

I don't _think_ we stated that a policy would get temporarily created (if you do, maybe you can point me to that comment/thread?). Instead it would created or not in ND depending on the signupQualifier (and it would assume that the BE has created the policy successfully). A policy should be created even before the account, so it's all happening before the user even signs in.

**Comment by marcaaron** [comment]
Please add manual QA steps for the accounting page flow.

**Comment by carlosmiceli** [comment]
Ah, cool, I interpreted that from this sentence:

> create a policy for now and then stop creating one after the backend creates one

But seems like we're on the same page now!

**Comment by Shahidullah-Muffakir** [comment]
Hi @carlosmiceli and @marcaaron,
Based on the discussions so far from here https://github.com/Expensify/App/issues/53326, we had assumed that the frontend changes in this PR would be merged simultaneously with the new backend changes. As a result, we completely skipped workspace creation for signups through OD in this PR.

However, from the discussions above, I understand that we may or may not create a workspace for OD signups in the frontend, depending on whether the backend has already created one. Therefore, we should still support workspace creation for OD signups on the frontend, and only skip it if the backend has already created a workspace for the user.

Did I understand this correctly? Thank you!

**Comment by carlosmiceli** [comment]
> depending on whether the backend has already created one

This is not totally accurate. The onboarding flow shouldn't need to check for existing workspaces created, it just needs to create a workspace or not depending on the `signupQualifier` value. That's what should decide if we create or skip creating a workspace during onboarding. If the account was created, and the `signupQualifier` is either `vsb` or `smb`, the onboarding flow should assume a workspace was properly created.

Do we agree @marcaaron?

**Comment by Shahidullah-Muffakir** [comment]
> Just for clarity, it **does not do this yet**. This PR will need to support the existing flow AND that change when it happens?

Yes, with this PR, we will still create a workspace for signups from ND that don’t have a signup qualifier, such as VSB or SMB.
If there is a signup qualifier, this PR will skip workspace creation since the backend will handle it.

> I think we have to unwind this part. Something needs the `policyID` IIRC (see [this PR](https://github.com/Expensify/App/pull/49161)). And, yes, the BE PRs will _eventually_ allow us to remove this check, but the plan was to deploy them after this PR is deployed.
> 
> So, App needs to exist in a state where it will maybe create a policy for now and then stop creating one after the backend creates one. At least, I thought that was the plan.
> 

We are still handling the policyID for ND signups here using `Welcome.setOnboardingPolicyID(policyID)`. For signups with VSB and SMB, I believe the backend should return the policyID via the `ONYXKEYS.ONBOARDING_POLICY_ID` onyx key when the workspace is created for those signups with a signup qualifier.

**Comment by Shahidullah-Muffakir** [comment]
@marcaaron 
> Just for clarity, it **does not do this yet**. This PR will need to support the existing flow AND that change when it happens?

Yes, with this PR, we will still create a workspace for signups from ND that don’t have a signup qualifier, such as VSB or SMB.
If there is a signup qualifier, this PR will skip workspace creation since the backend will handle it.

> I think we have to unwind this part. Something needs the `policyID` IIRC (see [this PR](https://github.com/Expensify/App/pull/49161)). And, yes, the BE PRs will _eventually_ allow us to remove this check, but the plan was to deploy them after this PR is deployed.
> 
> So, App needs to exist in a state where it will maybe create a policy for now and then stop creating one after the backend creates one. At least, I thought that was the plan.
> 

We are still handling the policyID for ND signups here using `Welcome.setOnboardingPolicyID(policyID)`. For signups with VSB and SMB, I believe the backend should return the policyID via the `ONYXKEYS.ONBOARDING_POLICY_ID` onyx key when the workspace is created for those signups with a signup qualifier.

**Comment by Shahidullah-Muffakir** [comment]
> > depending on whether the backend has already created one
> 
> This is not totally accurate. The onboarding flow shouldn't need to check for existing workspaces created, it just needs to create a workspace or not depending on the `signupQualifier` value. That's what should decide if we create or skip creating a workspace during onboarding. If the account was created, and the `signupQualifier` is either `vsb` or `smb`, the onboarding flow should assume a workspace was properly created.
> 
> Do we agree @marcaaron?

@luacmartins 
Great! In that case, we are doing exactly that in this PR.

File: src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx
Comment (Shahidullah-Muffakir) [reply src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx:98]: We moved the workspace creation logic here to `BaseOnboardingPurpose` because this page is only shown to users without VSB or SMB signup qualifiers. For users with VSB or SMB, this page is skipped, and the workspace creation won’t happen on the frontend.

I believe the workspace creation was temporarily moved to `BaseOnboardingEmployees` in your PR (https://github.com/Expensify/App/pull/50759) until the backend changes are ready. Now that we’ve moved the logic to BaseOnboardingPurpose, the workspace will only be created for users without VSB or SMB signup qualifiers, which is what we need to avoid duplicates.

**Comment by luacmartins** [comment]
>@luacmartins
Great! In that case, we are doing exactly that in this PR.

Was this meant for @carlosmiceli?

**Comment by marcaaron** [comment]
>We are still handling the policyID for ND signups here using Welcome.setOnboardingPolicyID(policyID). For signups with VSB and SMB, I believe the backend should return the policyID via the ONYXKEYS.ONBOARDING_POLICY_ID onyx key when the workspace is created for those signups with a signup qualifier.

Yes I agree with everything here. @carlosmiceli can address the `policyID` thing as the backend code has not be merged yet.

**Comment by Shahidullah-Muffakir** [comment]
> >@luacmartins
> Great! In that case, we are doing exactly that in this PR.
> 
> Was this meant for @carlosmiceli?

Apologies for the confusion, Yes, I meant to tag @carlosmiceli.

**Comment by carlosmiceli** [comment]
We realized that, since @Shahidullah-Muffakir makes a good point that we need to return the `policyID`, we should actually use that as a check to whether create a policy or not, instead of the `signupQualifier`. This simplifies the logic quite a bit. Can we implement that instead? I'll update the BE and let you know when it's merged. cc @thesahindia

**Comment by Shahidullah-Muffakir** [comment]
@carlosmiceli Currently, we have both checks:
`if (!onboardingPolicyID && !hasSignupQualifier && filteredPolicies.length === 0) {`

I’ll update the PR to remove the `hasSignupQualifier` check and rely only on the `onboardingPolicyID` check. That should be sufficient, Thank you.

**Comment by Shahidullah-Muffakir** [comment]
> We realized that, since @Shahidullah-Muffakir makes a good point that we need to return the `policyID`, we should actually use that as a check to whether create a policy or not, instead of the `signupQualifier`. This simplifies the logic quite a bit. Can we implement that instead? I'll update the BE and let you know when it's merged. cc @thesahindia

@carlosmiceli @thesahindia I removed the signupQualifier check for workspace creation, Thanks.

**Comment by carlosmiceli** [comment]
@Shahidullah-Muffakir great stuff, is this PR ready for review then?

**Comment by Shahidullah-Muffakir** [comment]
@carlosmiceli Yes, the PR is ready for review. Thank you!

**Comment by carlosmiceli** [comment]
Great, @thesahindia all yours!

**Comment by carlosmiceli** [comment]
@Shahidullah-Muffakir can you help me confirm that we can get the `signupBusinessPolicy` ID from the policy list sent with `onyxData` in `OpenApp`? I'm thinking that there's no point in adding a logic to check for onboarding policy ID during openApp given that it should only be used to avoid creating a policy while the onboarding hasn't been completed yet and for new users that are not coming from OD.

Since the user can't create a workspace until the onboarding is complete, there's no risk of having any other policy other than the personal and the signupBusinessPolicy in the list at this time.

Does this make sense? Am I missing something?

**Comment by Shahidullah-Muffakir** [comment]
@carlosmiceli If I understand correctly, you’re suggesting that we don’t need the `onboardingPolicyID` in `openApp` to avoid creating duplicate workspaces for OD signups. That’s correct because the current page (`BaseOnboardingPurpose`) is entirely skipped for OD signups, so a workspace won’t be created in that flow.


We still need `onboardingPolicyID` for other parts of the app, such as in the `completeOnboarding` function used in the tasks created during onboarding:

https://github.com/Expensify/App/blob/13286bec227f53a0b565e4fc1d54dc239dd3a78d/src/libs/actions/Report.ts#L3590-L3597

I couldn’t find the `signupBusinessPolicy` in the OpenApp response. If you’re suggesting that for OD signups, OpenApp will eventually return the `signupBusinessPolicy` and we then set it in the frontend as `onboardingPolicyID` using `Welcome.setOnboardingPolicyID(signupBusinessPolicy)`;, that approach should also work.

**Comment by carlosmiceli** [comment]
> I couldn’t find the signupBusinessPolicy in the `OpenApp` response.

It's not being sent that way inside `OnyxData` yet, but it's currently being returned under the `policy_` onyx collection (along with the personal policy) with `OpenApp`:

Screenshot 2024-12-19 at 12 49 34 PM

One of them should be `type: "personal"` and one should be `type: "team"` (the latter being the new one that we are creating when we create the account and that should decide if we should create one during onboarding or not). My thinking was that we could use this as a way to set `onboardingPolicyID`, what do you think?

**Comment by Shahidullah-Muffakir** [comment]
@carlosmiceli I believe we can use that approach.

However, currently, I can only see the policy of type `personal` being returned in the `OpenApp` endpoint, and not one of type `team`. If I can test this and confirm the team type policy is returned, I’ll make the necessary changes to update `onboardingPolicyID` using it and push the updates.

**Comment by carlosmiceli** [comment]
Right, we'll need to wait for the BE to be merged, it's not yet. But I wanted to confirm that this approach will work, this was the only thing holding back the BE in case we needed to send something different with OpenApp. I'll let you know as soon as it's deployed!

**Comment by carlosmiceli** [comment]
@Shahidullah-Muffakir ok, we realize that the FE should be merged first or otherwise we risk creating duplicate workspaces. However, the plan remains as we decided on the previous messages (getting the onboarding policy from the policy list). Can you proceed with the plan (hardcode values to test that it works) and once this is merged we can proceed with merging/testing the BE?

Let's make sure to test that this flow still creates a workspace for all the other existing cases, it should only skip creating one if the team policy is present in OnyxData after signing up in OD.

 Thanks for the patience and hard work!

**Comment by Shahidullah-Muffakir** [comment]
@carlosmiceli Sure, I’ll update and test the changes, Thanks.

**Comment by Shahidullah-Muffakir** [comment]
@carlosmiceli Added the changes to set `onboardingPolicyID` and `onboardingAdminsChatReportID` for OD signups based on the Policy returned from the BE.

**Comment by carlosmiceli** [comment]
Great! Is it ready for a review by @thesahindia ?

**Comment by Shahidullah-Muffakir** [comment]
> Great! Is it ready for a review by @thesahindia ?

Yes.

**Review by thesahindia — APPROVED** [review]
Looks good!

File: src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx
Comment (marcaaron) [reply src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx:97]: We have to fix this now because the FE needs will be merged first. Can you please return it to the correct spot?

File: src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx
Comment (marcaaron) [inline src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:69]: If we _were_ creating a policy here I'd assume that maybe we still need to create one until the BE changes are merged.

File: src/pages/OnboardingEmployees/BaseOnboardingEmployees.tsx
Comment (marcaaron) [inline src/pages/OnboardingEmployees/BaseOnboardingEmployees.tsx:70]: Could be wrong, but I'm pretty sure we still need this to complete the accounting step.

**Review by marcaaron — DISMISSED** [review]

**Comment by carlosmiceli** [comment]
@Shahidullah-Muffakir there are a couple of valid comments/concerns to address ☝️

File: src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx
Comment (Shahidullah-Muffakir) [reply src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx:69]: I've added the changes assuming the FE and BE will be merged at the same time. However, if we plan to merge the FE first, I will update the code accordingly.

**Comment by Shahidullah-Muffakir** [comment]
@carlosmiceli I've added the changes assuming FE and BE will be merged at the same time. However, if we merge FE first, I will update the code accordingly
Can you help clarify what `smb` is? It doesn't seem to be handled the same way as `vsb`. When will the `signup qualifier` have the value as `smb`? Thanks!

**Comment by carlosmiceli** [comment]
@Shahidullah-Muffakir VSB is "very small business" and SMB is "small business", but so far `smb` signups are not being redirected to ND. However, we may change that in the future, so we should cover both cases. We could also add a constant for `smb` like I assume we do for `vsb` if we don't have one yet.

**Comment by anmurali** [comment]
> When will the signup qualifier have the value as smb

image

When the user selects `Control expenses for a larger organization (10+ employees)`

**Comment by Shahidullah-Muffakir** [comment]
@carlosmiceli @anmurali Thanks for the clarification! 
Yes, we already have the constant for SMB as well.

**Comment by Shahidullah-Muffakir** [comment]
@marcaaron I updated the code as suggested, thanks.

**Comment by carlosmiceli** [comment]
@Shahidullah-Muffakir quick question: I see many checks for `isVsb`, but shouldn't we add a `isSmb` check in those places as well?

**Comment by Shahidullah-Muffakir** [comment]
@carlosmiceli The reason only `VSB` is mentioned is that the changes in this PR https://github.com/Expensify/App/issues/51750 were specifically added to handle workspace creation when the signup qualifier is VSB. Considering this, I did not modify the original logic. Additionally, since SMB signup qualifiers are not currently redirected to NewDot, I don’t think similar changes are needed for SMB.

**Comment by carlosmiceli** [comment]
@Shahidullah-Muffakir gotcha, just checking, thanks! Will review now.

**Review by carlosmiceli — APPROVED** [review]

**Comment by carlosmiceli** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] For any bug fix or new feature in this PR, I verified that sufficient [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) are included to prevent regressions in this flow.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/pages/OnboardingEmployees/BaseOnboardingEmployees.tsx
Comment (hungvu193) [inline src/pages/OnboardingEmployees/BaseOnboardingEmployees.tsx:35]: we filter out to get the first paid policy. If the user is invited to a policy with a user role, the paid policy can be the policy that the user is not an admin then the onboarding [API is called failure.](https://github.com/Expensify/App/issues/54849#issuecomment-2602987070).
https://github.com/Expensify/App/issues/54849

**Pull request merged**
